File: DataSet/y144/nomes.py
import re
with open('scopecontent.txt', 'r') as f:
    text = f.readlines()
    f.close()

def handle(line):
    line = line.strip()
    fex = r'[Ff]ilia[cç][aã]o:'
    # remove Filiação: from the beginning
    line = re.sub(fex, '', line)
    line = line.split('.')[0]
    ns = []
    for n in re.split(r'\be\b', line):
        n = n
        n = n.split(',')
        n[0] = '<a href="registo/Inquirição de genere de'+n[0]+'">' + n[0] + '</nome>'
        ns.append(','.join(n))
    return ' e '.join(ns)+'\n'


fo = open('SCout.txt', 'w')
flag = True
text = ['Filiação: Gracia Maria,Solteira. Natural e/ou residente em ARCOZELO,Sao Mamede, actual concelho de BARCELOS e distrito (ou país) Braga.']
for l in text:
    if l != '\n':
        fo.write(handle(l))
        #fo.write(handle(l))

File: DataSet/y144/nomes.js

var fs = require('fs');
var text = fs.readFileSync('scopecontent.txt','utf8');

// open file for writing


function handle(line){
    line = line.trim()
    var fex = /[Ff]ilia[cç][aã]o:/g
    // remove Filiação: from the beginning
    line = line.split(fex)
    if (line.length < 2)
        l2 = line[0]
    else
        l2 = line[1].split('.')
    first = l2[0]
    ns = []
    for(n of first.split(/\be\b/g)){
        var n1 = n.split(',')
        n1[0] = '<a href="registo/Inquirição de genere de '+n1[0].trim()+'">' + n1[0] + '</nome>'
        ns.push(n1.join(','))
    }
    l2[0] = ns.join('e')
    line[1] = l2.join('.')
    line = line.join('Filiação:')
    return line
}
